purchase_incoterm/models/purchase_order.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.onchange('partner_id')
+    def _onchange_partner(self):
+        if self.partner_id:
+            self.incoterm_id = self.partner_id.purchase_incoterm_id
+        else:
+            self.incoterm_id = False
```

Comment:
There is already a `onchange_partner_id` on `purchase.order`, better to re-use it and simplify the implementation:
```suggestion
    def onchange_partner_id(self):
        res = super().onchange_partner_id()
        self.incoterm_id = self.partner_id.purchase_incoterm_id
        return res
```

Reply:
Thanks for reivew, branch updated